tests/subsys/profiler/src/main.c

```diff
@@ -0,0 +1,149 @@
+#define EXAMPLE_STRING "example string"
+
+static uint16_t no_data_event_id;
+static uint16_t data_event_id;
+static uint16_t big_event_id;
+
+static void profile_no_data_event(void)
+{
+	struct log_event_buf buf;
+
+	profiler_log_start(&buf);
+	profiler_log_send(&buf, no_data_event_id);
+}
+
+static void profile_data_event(uint32_t val1)
+{
+	struct log_event_buf buf;
+
+	profiler_log_start(&buf);
+	profiler_log_encode_uint32(&buf, val1);
+	profiler_log_send(&buf, data_event_id);
+}
+
+static void profile_big_event(uint32_t val1, int32_t val2, uint16_t val3,
+			      int16_t val4, uint8_t val5, int8_t val6,
+			      const char *string)
+{
+	struct log_event_buf buf;
+
+	profiler_log_start(&buf);
+	profiler_log_encode_uint32(&buf, val1);
+	profiler_log_encode_int32(&buf, val2);
+	profiler_log_encode_uint16(&buf, val3);
+	profiler_log_encode_int16(&buf, val4);
+	profiler_log_encode_uint8(&buf, val5);
+	profiler_log_encode_int8(&buf, val6);
+	profiler_log_encode_string(&buf, string);
+	profiler_log_send(&buf, big_event_id);
+}
+
+static void register_profiler_events(void)
+{
+	static const char * const data_names[] = {"value1"};
+	static const enum profiler_arg data_types[] = {PROFILER_ARG_U32};
+	static const char * const big_event_data_names[] = {"value1", "value2", "value3",
+							  "value4", "value5", "value6",
+							  "string"};
+	static const enum profiler_arg big_event_data_types[] = {PROFILER_ARG_U32,
+								 PROFILER_ARG_S32,
+								 PROFILER_ARG_U16,
+								 PROFILER_ARG_S16,
+								 PROFILER_ARG_U8,
+								 PROFILER_ARG_S8,
+								 PROFILER_ARG_STRING};
+	no_data_event_id = profiler_register_event_type("no data event", NULL,
+							NULL, 0);
+	data_event_id = profiler_register_event_type("data event", data_names,
+						     data_types, 1);
+	big_event_id = profiler_register_event_type("big event", big_event_data_names,
+						    big_event_data_types, 7);
+}
+
+void test_init(void)
+{
+	zassert_ok(profiler_init(), "Error when initializing");
+	register_profiler_events();
+}
+
+static void test_performance1(void)
+{
+	uint32_t start_time;
+	uint32_t elapsed_ticks;
+	uint32_t elapsed_time_us;
+
+	/* Profiling no data event */
+	start_time = k_cycle_get_32();
+	for (size_t i = 0; i < PROFILED_EVENTS_NB; i++) {
+		profile_no_data_event();
+	}
+	elapsed_ticks = (uint32_t) k_cycle_get_32() - start_time;
+	elapsed_time_us = (uint32_t) elapsed_ticks * MS_PER_TIMESTAMP_TICK * 1000;
+	printk("Logged %s events with no data.\nElapsed time [us]: %d\n",
+	       STRINGIFY(PROFILED_EVENTS_NB), elapsed_time_us);
+}
+
+static void test_performance2(void)
+{
+	uint32_t start_time;
+	uint32_t elapsed_ticks;
+	uint32_t elapsed_time_us;
+
+	/* Profiling data event */
+	start_time = k_cycle_get_32();
+	for (size_t i = 0; i < PROFILED_EVENTS_NB; i++) {
+		profile_data_event((uint32_t) i);
+	}
+	elapsed_ticks = (uint32_t) k_cycle_get_32() - start_time;
+	elapsed_time_us = (uint32_t) elapsed_ticks * MS_PER_TIMESTAMP_TICK * 1000;
```

Comment:
Use functions provided by Zephyr for simplicity:
https://docs.zephyrproject.org/apidoc/latest/time__units_8h.html